test/cql-pytest/test_compactionstats.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import pytest
+
+import nodetool
+from util import new_test_table
+
+
+def get_compaction_stats(cql, table):
+    ks, cf = table.split('.')
+    stats = nodetool.compactionstats(cql)
```

Comment:
This assumes the `if has_rest_api(cql):` branch is taken.
Why is the `else` branch needed? 
If it's ever taken, the test won't work because 1. it return a string instead of dict, and 2. `compactionstats` return more than `pending_tasks_by_table`.

Reply:
Please make sure the test also works with Cassandra (test/cql-pytest/run-cassandra - see test/cql-pytest/README.md for instructions if you've never done this) which uses actual nodetool.  Alternatively, if this is a test that ONLY works in Scylla for some reason, please explain the reason in a comment above the test and add the "scylla_only" fixture.

If the format of the REST API's output is different from nodetool, we can't pretend that nodetool.compatstats() can just call the REST API, and maybe it needs to transform its output before returning.

Alternatively, if as @cezarmoise suggested, this test has nothing to do with nodetool and actually only checks the REST API, let's drop the nodetool facade and just say we are testing the REST API - and then obviously it will be scylla_only. You can also move it to test/rest_api, where all tests are Scylla-only. See also my question above if perhaps test/rest_api/test_task_manager.py is the right place for this test.

Reply:
@bhalevy but does it work on Cassandra @cezarmoise suggested the output format is different in those two commands (REST API and "nodetool"). If this is true, then this nodetool.get_compaction_stats() implementation is wrong. But if it's not true, then I didn't understand what @cezarmoise said.